### PR TITLE
fix: Mutex never gets released on refresh function of AccountTrackerController

### PR DESCRIPTION
**Pull request by tommasini**
## Explanation

Currently the mutex do not get released unless there is an error thrown by the `getBalanceFromChain` function
Fixed by move the release of the murex to the finally block. It was also removed unnecessary catch block.

Now the mutex gets released when the logic of refresh runs successfully 

### Fixed
- Ensure mutex is released when refresh succeeds
  - Previously the `refresh` method would remain locked indefinitely after it was run successfully. The mutex was only released upon failure.

### `@metamask/assets-controllers`

- **FIXED**: Mutex released on the finally block of refresh function

## Checklist

- [ ] I've updated the test suite for new or updated code as appropriate
- [ ] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [ ] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

**Review by Gudahtt — COMMENTED** [review]
The diff looks great! Would you mind filling out the rest of the PR template?

And for the change entry, I would suggest rewording it to describe how the behavior has changed, rather than what we did to change it.

e.g. something like:

```
### Fixed
- Ensure mutex is released when refresh succeeds
  - Previously the `refresh` method would remain locked indefinitely after it was run successfully. The mutex was only released upon failure.
```

**Comment by tommasini** [comment]
Added! By mistake I left my description under the comments of the PR template

**Review by legobeat — APPROVED** [review]

**Review by legobeat — APPROVED** [review]
LGTM!
